dataset/variable_reduction.cpp

```diff
@@ -65,8 +60,10 @@ Variable nanmean(const VariableConstView &var, const Dim dim,
                  const MasksConstView &masks) {
   using variable::isfinite;
   if (const auto mask_union = irreducible_mask(masks, dim)) {
-    const auto count = sum(applyMask(isfinite(var), mask_union), dim);
-    return nanmean_impl(applyMask(var, mask_union), dim, count);
+    const auto count =
+        sum(scipp::variable::masked_to_zero(isfinite(var), mask_union), dim);
```

Comment:
You don't need the `scipp::variable` here or in any of the other cases. ADL will take care of the test.